### ISSUE TITLE
[Feature Request] Maximum limit

### ISSUE THREAD

**Issue by jean1880** [balderdashy/sails#3295]
## Summary (TL;DR)
- Ability to set a hard limit on data returns that cannot be overridden by a find query

## Description
This may just be me missing something but, Sails has a default limit, however in a find query, you can override the limit, specifying any value the user wishes. This is a potential security issue if you have a table of hundreds of thousands of rows, allowing a user to start making a series of CPU intensive calls for massive listing dumps. I haven't been able to find any setting to set a hard limit, and the only way around this, that I am aware of, would be to write your own find statements. This isn't necessarily a huge task, but it would be nice if we could simply set a global setting value to prevent massive data dumps. I don't foresee this being an issue for any use case scenarios, as the default limit would act as the standard return, and the max limit would at as a safeguard, limiting a call to a reasonable size.

**Comment by jamesjwarren** [comment]
Currently it would be possible to enforce this limit in a policy, by checking the limit param and ensuring it does not exceed a pre-defined limit. However I agree it would be nice to have a configurable option for the overall max limit.

**Comment by sailsbot** [comment]
Thanks for posting, @jean1880.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
